softwareconfiguration/hasco/src/main/java/hasco/variants/forwarddecomposition/twophase/TwoPhaseHASCO.java

```diff
@@ -168,9 +168,10 @@ public AlgorithmEvent nextWithException() throws InterruptedException, Algorithm
 			}
 			this.secondsSpentInPhase1 = (int) Math.round(System.currentTimeMillis() - this.timeOfStart / 1000.0);
 
+			/* if there is no candidate, and the remaining time is very small, throw an AlgorithmTimeoutedException */
 			this.logger.info("HASCO has finished. {} solutions were found.", this.phase1ResultQueue.size());
-			if (this.phase1ResultQueue.isEmpty()) {
-				throw new NoSuchElementException("No classifier could be built within the given timeout.");
+			if (this.phase1ResultQueue.isEmpty() && this.getRemainingTimeToDeadline().seconds() < 10) {
```

Comment:
hard coded constant of 10 seconds for phase1 result queue.
If the task is very simple it could be reasonable to believe it was solvable within 10 seconds.